### DISCUSSION TITLE
Body updates example code in the documentation only works for flat models

### DISCUSSION THREAD

**Discussion by gnu-lorien** [fastapi/fastapi#8573] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class SubSubModel(pydantic.BaseModel):
    first: str = "first"
    second: str = "second"


class SubModel(pydantic.BaseModel):
    base: SubSubModel = pydantic.Field(default_factory=lambda: SubSubModel())
    mapping: typing.Dict[str, SubSubModel] = pydantic.Field(default_factory=dict)


class TestModel(pydantic.BaseModel):
    sub: SubModel = pydantic.Field(default_factory=lambda: SubModel())
```

### Description

In the tutorial for Body Updates here https://fastapi.tiangolo.com/tutorial/body-updates/ it gives examples of using pydantic's `copy(update=` in order to handle partial updates of model data. This method only works when the model is flat. If you have any submodels or mapping types then this method will both wipe out previous data and leave you with an invalid pydantic model. I have also filed an issue with pydantic about this, but, in the mean time, it's a gotcha in the FastAPI documentation that left me fairly frustrated to deal with. Perhaps the documentation could be updated to specifically call out that this method doesn't work with mapping types or submodels?

The issue as I've described it on the pydantic side: https://github.com/samuelcolvin/pydantic/issues/3785

### Operating System

Linux

### Operating System Details

CentOS

### FastAPI Version

0.73.0

### Python Version

Python 3.9.0

### Additional Context

This is more of a documentation enhancement request.

**Comment by YuriiMotov** [comment]
Current version works fine
